.github/workflows/ci.yml

```diff
@@ -320,32 +354,47 @@ jobs:
         include:
           - target: i686-unknown-linux-gnu
             rustflags: --cfg tokio_taskdump
-          - target: arm-unknown-linux-gnueabihf
+          - target: armv5te-unknown-linux-gnueabi
           - target: armv7-unknown-linux-gnueabihf
           - target: aarch64-unknown-linux-gnu
             rustflags: --cfg tokio_taskdump
 
           # Run a platform without AtomicU64 and no const Mutex::new
-          - target: arm-unknown-linux-gnueabihf
+          - target: armv5te-unknown-linux-gnueabi
             rustflags: --cfg tokio_no_const_mutex_new
     steps:
       - uses: actions/checkout@v3
       - name: Install Rust stable
-        uses: dtolnay/rust-toolchain@master
+        uses: dtolnay/rust-toolchain@stable
         with:
           toolchain: ${{ env.rust_stable }}
           target: ${{ matrix.target }}
-      - name: Install cross
-        uses: taiki-e/install-action@cross
-      # First run with all features (including parking_lot)
-      - run: cross test -p tokio --all-features --target ${{ matrix.target }} --tests
+
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-nextest
+      - uses: taiki-e/setup-cross-toolchain-action@v1
+        with:
+          target: ${{ matrix.target }}
+
+      - uses: Swatinem/rust-cache@v2
+      - name: First tests run with all features (including parking_lot)
+        run: |
+          set -euxo pipefail
+          cargo nextest run -p tokio --all-features --target ${{ matrix.target }}
+          cargo test --doc -p tokio --all-features --target ${{ matrix.target }}
```

Comment:
Hmm. nextest+setup-cross-toolchain-action ([log](https://github.com/tokio-rs/tokio/actions/runs/4981204722/jobs/8915183782)) seems to be slower than cross ([log](https://github.com/tokio-rs/tokio/actions/runs/4980067925/jobs/8912721614)).

I guess it's one of the following two things:

- The parallelization performed by libtest is causing a problem with QEMU, and setting `RUST_TEST_THREADS=1` environment variable fixes the problem.
- The parallelization performed by nextest is causing a problem with QEMU, and using `cargo test` instead of `cargo nextest run` fixes the problem.

(In the current CI run, setup-cross-toolchain-action is a bit [slow](https://github.com/tokio-rs/tokio/actions/runs/4981204722/jobs/8915183928) because it depends on the state of Azure's APT mirror used in GitHub Actions runners, but it is usually [much faster](https://github.com/taiki-e/portable-atomic/actions/runs/4969849868/jobs/8893446429).)

Reply:
> The parallelization performed by nextest is causing a problem with QEMU, and using `cargo test` instead of `cargo nextest` run fixes the problem.

AFAIK `cargo-nextest` runs each test in a separate process, so this should not be a problem.

It's likely `cargo test --doc` is a problem, passing `--test-threads=1` to it might fix the problem.

Reply:
I've added `-- --test-threads=1` to all invocation of `cargo-test` in `no-atomics` and `cross-test*`.

Reply:
I've also added in to job `asan` where I tried using `cargo-nextest` again.